src/vaults/AsyncRequests.sol

```diff
@@ -629,4 +633,11 @@ contract AsyncRequests is BaseInvestmentManager, IAsyncRequests {
 
         return VaultPricingLib.calculatePrice(shareToken, shares, vaultDetails.asset, vaultDetails.tokenId, assets);
     }
+
+    function _mapPoolId(uint64 poolId) internal view returns (uint64 mappedPoolId) {
+        // TODO: update legacyCentrifugeId and add all pools before deployment
+        uint16 legacyCentrifugeId = 1;
+        if (poolId == 4139607887) return newPoolId(legacyCentrifugeId, 1).raw();
```

Comment:
Since pool creation is permissioned, we c an just ensure the legacy pool(s) are created in the order specified here. This removes the need for storage lookups, so doesn't add extra gas cost.